tests/next_tests/integration_tests/feature_tests/ffront_tests/test_execution.py

```diff
@@ -38,296 +38,255 @@
         qc = qc_in + carry + scalar
         return qc
 
-    @program(backend=fieldview_backend)
+    @program
     def scan_scalar(qc: Field[[IDim, KDim], float], scalar: float):
         _scan_scalar(qc, scalar, out=qc)
 
-    scan_scalar(qc, scalar, offset_provider={})
-    assert np.allclose(np.asarray(qc), expected)
+    qc = cases.allocate(cartesian_case, scan_scalar, "qc").strategy(cases.zeros)()
+    scalar = cases.allocate(cartesian_case, scan_scalar, "scalar").strategy(
+        lambda dtype, shape: np.dtype(dtype).type(1)
+    )()
+    ksize = cartesian_case.default_sizes[KDim]
+    expected = np.full((ksize, ksize), np.arange(start=1, stop=11, step=1).astype(float64))
 
+    cases.verify(cartesian_case, scan_scalar, qc, scalar, nopass_out=qc, ref=expected)
 
-def test_tuple_scalar_scan(fieldview_backend):
-    if fieldview_backend in [gtfn_cpu.run_gtfn, gtfn_cpu.run_gtfn_imperative]:
-        pytest.skip("Scalar tuple arguments are not supported in gtfn yet.")
 
-    size = 10
-    KDim = Dimension("K", kind=DimensionKind.VERTICAL)
-    qc = np_as_located_field(IDim, KDim)(np.zeros((size, size)))
-    tuple_scalar = (1.0, (1.0, 0.0))
-    expected = np.full((size, size), np.arange(start=1, stop=11, step=1).astype(float64))
+def test_tuple_scalar_scan(cartesian_case):  # noqa: F811 # fixtures
+    if cartesian_case.backend in [gtfn_cpu.run_gtfn, gtfn_cpu.run_gtfn_imperative]:
+        pytest.skip("Scalar tuple arguments are not supported in gtfn yet.")
 
     @scan_operator(axis=KDim, forward=True, init=0.0)
     def _scan_tuple_scalar(
         state: float, qc_in: float, tuple_scalar: tuple[float, tuple[float, float]]
     ) -> float:
         return (qc_in + state + tuple_scalar[1][0] + tuple_scalar[1][1]) / tuple_scalar[0]
 
-    @field_operator(backend=fieldview_backend)
+    @field_operator
     def scan_tuple_scalar(
         qc: Field[[IDim, KDim], float], tuple_scalar: tuple[float, tuple[float, float]]
     ) -> Field[[IDim, KDim], float]:
         return _scan_tuple_scalar(qc, tuple_scalar)
 
-    scan_tuple_scalar(qc, tuple_scalar, out=qc, offset_provider={})
-    assert np.allclose(np.asarray(qc), expected)
+    qc = cases.allocate(cartesian_case, scan_tuple_scalar, "qc").strategy(cases.zeros)()
+    tuple_scalar = (1.0, (1.0, 0.0))
+    ksize = cartesian_case.default_sizes[KDim]
+    expected = np.full((ksize, ksize), np.arange(start=1, stop=11, step=1).astype(float64))
 
+    cases.verify(cartesian_case, scan_tuple_scalar, qc, tuple_scalar, out=qc, ref=expected)
 
-def test_astype_int(fieldview_backend):
-    size = 10
-    b_float_64 = np_as_located_field(IDim)(np.full((size,), fill_value=1.5, dtype=np.float64))
-    c_int64 = np_as_located_field(IDim)(np.ones((size,), dtype=np.int64))
-    out_int_64 = np_as_located_field(IDim)(np.zeros((size,), dtype=np.int64))
 
-    @field_operator(backend=fieldview_backend)
+def test_astype_int(cartesian_case):  # noqa: F811 # fixtures
```

Comment:
I think we should put the 3 following tests into a parametrized one. Could you try if that's compatible with the new infrastructure? (Disclaimer: not sure if its easily doable without the new infrastructure.

Reply:
I don't currently see a way to do it without extending the toolchain. `astype(x, <type_builtin>)` does not look into `<type_builtin>` to see if it is a variable containing a type builtin. The toolchain only looks at the name.

Reply:
Everything else should work though:
 ✅  field type annotations parametrized by dtype
 ✅  allocating the output fields parametrized by dtype
 ✅  reference parametrized by dtype
 ❌  field operator with `astype` parametrized by dtype